metron-platform/metron-common/src/main/java/org/apache/metron/common/writer/BulkWriterComponent.java

```diff
@@ -103,10 +115,21 @@ public void write( String sensorType
     }
     messageList.add(message);
 
-    if (tupleList.size() < batchSize) {
-      sensorTupleMap.put(sensorType, tupleList);
-      sensorMessageMap.put(sensorType, messageList);
-    } else {
+    if(configurations.getGlobalConfig()!=null&&configurations.getGlobalConfig().get(Constants.TIME_FLUSH_FLAG)!=null)
```

Comment:
I'd recommend saving off the global config in a member variable and reusing that reference.  It's possible the configuration will get updated between calls to `configurations.getGlobalConfig()` in zookeeper and therefore you'll get two separate configs.